src/utils.ts

```diff
@@ -120,9 +155,44 @@ export function formatBytes(bytes: number) {
   return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
 }
 
-export interface FrontMatterResult {
-  frontMatter: string | null;
-  body: string | null;
+export function getLocalizedValue(doc: Document, item: any, key: string) {
+  return doc.locale ? item[`${key}@${doc.locale.id}`] || item[key] : item[key];
+}
+
+export function interpolate(pod: Pod, string: string, params: any) {
+  // Cache created functions to avoid memory leak.
+  const names = Object.keys(params);
+  const vals = Object.values(params);
+  const key = `${string}:${names}`;
+  if (pod.cache.interpolations[key]) {
+    return pod.cache.interpolations[key](...vals);
+  }
+  const func = new Function(...names, `return \`${string}\`;`);
+  pod.cache.interpolations[key] = func;
+  return func(...vals);
+}
+
+export function localizeData(data: any, locale: Locale) {
+  if (!data) {
+    return data;
+  }
+
+  if (DataType.isObject(data)) {
+    const newData: any = {};
+    for (const key of Object.keys(data)) {
+      newData[key] = localizeData(data[key], locale);
+    }
+    return newData;
+  } else if (DataType.isArray(data)) {
```

Comment:
A general pattern I prefer is if the `if()` statement above always has a `return`, then use `if` in the next line instead of `else if`.

E.g.:

```javascript
if (...) {
  // ...
  return something;
}

if (...) {
  /// ...
}
```

Reply:
That is a good idea, I'll try and be aware of that.